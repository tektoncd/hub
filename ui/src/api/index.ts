import axios from 'axios';
import { API_URL } from '../config/constants';
import { ICategory } from '../store/category';
import { IResource, IVersion } from '../store/resource';

interface Token {
  token: string;
  refreshInterval: string;
  expiresAt: number;
}

interface TokenData {
  access: Token;
  refresh: Token;
}

export interface AuthResponse {
  data: TokenData;
}

export interface Rating {
  rating: number;
}

export interface Api {
  categories(): Promise<ICategory>;
  resources(): Promise<IResource>;
  resourceVersion(resourceId: number): Promise<IVersion>;
  versionUpdate(versionId: number): Promise<IVersion>;
  authentication(authCode: string): Promise<AuthResponse>;
  readme(rawURL: string): Promise<string>;
  yaml(rawURL: string): Promise<string>;
  getRating(resourceId: number, token: string): Promise<Rating>;
  setRating(resourceId: number, token: string, rating: number): Promise<void | null>;
}

export class Hub implements Api {
  async categories() {
    try {
      return axios.get(`${API_URL}/categories`).then((response) => response.data);
    } catch (err) {
      return err.response;
    }
  }

  async resources() {
    try {
      return axios.get(`${API_URL}/resources`).then((response) => response.data);
    } catch (err) {
      return err.response;
    }
  }

  async authentication(authCode: string) {
    try {
      return axios
        .post(`${API_URL}/auth/login?code=${authCode}`)
        .then((response) => response.data)
        .catch((err) => Promise.reject(err.response));
    } catch (err) {
      return err;
    }
  }

  async resourceVersion(resourceId: number) {
    try {
      return axios
        .get(`${API_URL}/resource/${resourceId}/versions`)
        .then((response) => response.data);
    } catch (err) {
      return err.response;
    }
  }

  async versionUpdate(versionId: number) {
    try {
      return axios
        .get(`${API_URL}/resource/version/${versionId}`)
        .then((response) => response.data);
    } catch (err) {
      return err.response;
    }
  }

  async readme(rawURL: string) {
    try {
      const URL = rawURL.substring(0, rawURL.lastIndexOf('/') + 1);
      return axios.get(`${URL}/README.md`).then((response) => response.data);
    } catch (err) {
      return err.response;
    }
  }

  async yaml(rawURL: string) {
    try {
      const newLine = '\n';
      return axios.get(`${rawURL}`).then((response) => '```yaml' + newLine + response.data);
    } catch (err) {
      return err.response;
    }
  }

  async getRating(resourceId: number, token: string) {
    try {
      return axios
        .get(`${API_URL}/resource/${resourceId}/rating`, {
          headers: {
            Authorization: `Bearer ${token}`
          }
        })
        .then((response) => response.data)
        .catch((err) => Promise.reject(err.response));
    } catch (err) {
      return err;
    }
  }

  async setRating(resourceId: number, token: string, rating: number) {
    try {
      return axios
        .put(
          `${API_URL}/resource/${resourceId}/rating`,
          { rating: rating },
          {
            headers: {
              Authorization: `Bearer ${token}`
            }
          }
        )
        .then((response) => response.data)
        .catch((err) => Promise.reject(err.response));
    } catch (err) {
      return err;
    }
  }
}
