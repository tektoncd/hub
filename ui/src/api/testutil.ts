import { Api } from './';
import * as fs from 'fs';
import { ICategory } from '../store/category';
import { IResource, IVersion } from '../store/resource';

export class FakeHub implements Api {
  dataDir: string;

  constructor(dataDir: string) {
    this.dataDir = dataDir;
  }

  async categories() {
    const data = `${this.dataDir}/categories.json`;
    const ret = () => JSON.parse(fs.readFileSync(data).toString());
    return new Promise<ICategory>((resolve) => {
      setTimeout(() => resolve(ret()), 1000);
    });
  }

  async resources() {
    const data = `${this.dataDir}/resources.json`;

    const ret = () => JSON.parse(fs.readFileSync(data).toString());
    return new Promise<IResource>((resolve) => {
      setTimeout(() => resolve(ret()), 1000);
    });
  }

  async resourceVersion(resourceId: number) {
    const data = `${this.dataDir}/resource_versions_${resourceId}.json`;

    const ret = () => JSON.parse(fs.readFileSync(data).toString());
    return new Promise<IVersion>((resolve) => {
      setTimeout(() => resolve(ret()), 1000);
    });
  }

  async versionUpdate(versionId: number) {
    const data = `${this.dataDir}/version_details_${versionId}.json`;

    const ret = () => JSON.parse(fs.readFileSync(data).toString());
    return new Promise<IVersion>((resolve) => {
      setTimeout(() => resolve(ret()), 1000);
    });
  }
}
