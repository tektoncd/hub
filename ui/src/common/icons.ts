export enum Icons {
  Unknown = 'unknown',
  Cat = 'cat',
  Certificate = 'certificate',
  User = 'user',
  Build = 'build',
  Domain = 'domain'
}
