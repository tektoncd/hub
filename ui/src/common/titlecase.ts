/**
 * To convert the first letter of the input word to Upper Case
 */
export const titleCase = (filterName: string) => {
  return filterName.replace(/\w\S*/g, function (word) {
    return word.charAt(0).toUpperCase() + word.substr(1).toLowerCase();
  });
};
