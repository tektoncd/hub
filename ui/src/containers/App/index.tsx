import React from 'react';
import { observer } from 'mobx-react';
import '@patternfly/react-core/dist/styles/base.css';
import { Page } from '@patternfly/react-core';
import { BrowserRouter as Router, Route, Routes, Navigate } from 'react-router-dom';
import HomePage from '../../components/HomePage';
import ResourceDetailsPage from '../../components/ResourceDetailsPage';
import Header from '../../containers/Header';
import Footer from '../../components/Footer';
import ParseUrl from '../ParseUrl';
import { createProvider } from '../../store/root';
import './App.css';

const Provider = createProvider();

const App: React.FC = observer(() => {
  return (
    <Provider>
      <Router>
        <ParseUrl />
        <Page header={<Header />} className="hub-page">
          <Routes>
            <Route path="/login" element={<Navigate to="/" replace />} />
            <Route path="/" element={<HomePage />}></Route>
            <Route path="/:catalog/:kind/:name/:version?" element={<ResourceDetailsPage />}></Route>
          </Routes>
          <Footer />
        </Page>
      </Router>
    </Provider>
  );
});

export default App;
