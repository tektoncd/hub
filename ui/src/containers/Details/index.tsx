import React from 'react';
import { useObserver } from 'mobx-react';
import { useParams } from 'react-router-dom';
import { Spinner } from '@patternfly/react-core';
import { useMst } from '../../store/root';
import BasicDetails from '../BasicDetails';
import Description from '../../components/Description';
import { assert } from '../../store/utils';

const Details: React.FC = () => {
  const { resources, user } = useMst();
  const { name } = useParams();

  const resourceDetails = () => {
    resources.versionInfo(name);
    resources.loadReadme(name);
    resources.loadYaml(name);

    const resource = resources.resources.get(name);
    assert(resource);
    user.getRating(resource.id);
  };

  const scrollToTop = () => {
    const scroller = document.querySelector('main');
    assert(scroller);
    if (scroller) scroller.scrollTo(0, 0);
  };

  return useObserver(() =>
    resources.resources.size === 0 ? (
      <Spinner className="hub-spinner" />
    ) : (
      <React.Fragment>
        {resourceDetails()}
        {scrollToTop()}
        <BasicDetails />
        <Description name={name} />
      </React.Fragment>
    )
  );
};
export default Details;
