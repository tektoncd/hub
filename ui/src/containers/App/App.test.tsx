import React from 'react';
import { mount } from 'enzyme';
import { FakeHub } from '../../api/testutil';
import { createProviderAndStore, createProvider } from '../../store/root';
import LeftPane from '../../components/LeftPane';
import App from '.';
import { FakeDate, ActualDate } from '../../common/testutils';
import Details from '../Details';

const TESTDATA_DIR = `src/store/testdata`;
const api = new FakeHub(TESTDATA_DIR);
const { Provider } = createProviderAndStore(api);

// Assign a dummy date to global Date inorder to avoid snapshaot updation
FakeDate();

// Assign current Date once test is over
ActualDate();

describe('App', () => {
  it('should render the component correctly and match the snapshot', (done) => {
    const app = mount(
      <Provider>
        <App />
      </Provider>
    );

    expect(app.debug()).toMatchSnapshot();
    done();
  });

  it('should find the component and match the count', () => {
    const Provider = createProvider();
    const component = mount(
      <Provider>
        <App />
      </Provider>
    );
    expect(component.find(LeftPane).length).toEqual(1);
  });
});
