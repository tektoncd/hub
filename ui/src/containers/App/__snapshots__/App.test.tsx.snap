// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render the component correctly and match the snapshot 1`] = `
"<Provider>
   
  <Memo(wrappedComponent)>
    <Provider>
       
      <BrowserRouter>
        <Router history={{...}}>
          <ParseUrl>
             
          </ParseUrl>
          <Page header={{...}} className=\\"hub-page\\" isManagedSidebar={false} isBreadcrumbWidthLimited={false} defaultManagedSidebarIsOpen={true} onPageResize={[Function: onPageResize]} mainTabIndex={-1} isNotificationDrawerExpanded={false} onNotificationDrawerExpand={[Function: onNotificationDrawerExpand]}>
            <div className=\\"pf-c-page hub-page\\">
              <Memo(wrappedComponent)>
                <PageHeader logo={{...}} headerTools={{...}}>
                  <header role={[undefined]} className=\\"pf-c-page__header\\">
                    <div className=\\"pf-c-page__header-brand\\">
                      <a className=\\"pf-c-page__header-brand-link\\">
                        <Brand src=\\"logo.png\\" alt=\\"Tekton Hub Logo\\" onClick={[Function: homePage]}>
                          <img onClick={[Function: homePage]} className=\\"pf-c-brand\\" src=\\"logo.png\\" alt=\\"Tekton Hub Logo\\" />
                        </Brand>
                      </a>
                    </div>
                    <PageHeaderTools>
                      <div className=\\"pf-c-page__header-tools\\">
                        <Grid>
                          <div className=\\"pf-l-grid\\">
                            <GridItem span={10}>
                              <div className=\\"pf-l-grid__item pf-m-10-col\\">
                                <Memo(wrappedComponent)>
                                  <ForwardRef type=\\"search\\" value=\\"\\" onChange={[Function: onSearchChange]} onKeyPress={[Function: onSearchKeyPress]} aria-label=\\"text input example\\" placeholder=\\"Search for resources...\\" spellCheck=\\"false\\" className=\\"hub-search\\">
                                    <TextInputBase type=\\"search\\" value=\\"\\" onChange={[Function: onSearchChange]} onKeyPress={[Function: onSearchKeyPress]} aria-label=\\"text input example\\" placeholder=\\"Search for resources...\\" spellCheck=\\"false\\" className=\\"hub-search\\" innerRef={{...}} isRequired={false} validated=\\"default\\" isDisabled={false} isReadOnly={false} isLeftTruncated={false}>
                                      <input onKeyPress={[Function: onSearchKeyPress]} aria-label=\\"text input example\\" placeholder=\\"Search for resources...\\" spellCheck=\\"false\\" onFocus={[Function (anonymous)]} onBlur={[Function (anonymous)]} className=\\"pf-c-form-control hub-search\\" onChange={[Function (anonymous)]} type=\\"search\\" value=\\"\\" aria-invalid={false} required={false} disabled={false} readOnly={false} />
                                    </TextInputBase>
                                  </ForwardRef>
                                </Memo(wrappedComponent)>
                              </div>
                            </GridItem>
                            <GridItem span={1} onClick={[Function: onClick]} className=\\"header-search-hint\\">
                              <div className=\\"pf-l-grid__item pf-m-1-col header-search-hint\\" onClick={[Function: onClick]}>
                                <Icon id=\\"help\\" size=\\"sm\\" label=\\"search-tips\\">
                                  <HelpIcon size=\\"sm\\" label=\\"search-tips\\" color=\\"currentColor\\" noVerticalAlign={false}>
                                    <svg style={{...}} fill=\\"currentColor\\" height=\\"1em\\" width=\\"1em\\" viewBox=\\"0 64 1024 1024\\" aria-labelledby={{...}} aria-hidden={true} role=\\"img\\" label=\\"search-tips\\">
                                      <path d=\\"M512.059-73.143c-282.338 0-512.059 229.673-512.059 512.025 0 282.235 229.721 511.975 512.059 511.975 282.281 0 511.941-229.735 511.941-511.975 0.005-282.352-229.659-512.025-511.941-512.025zM512.059 826.523c-213.826 0-387.728-173.856-387.728-387.643 0-213.89 173.904-387.694 387.728-387.694 213.717 0 387.671 173.803 387.671 387.694 0.005 213.785-173.957 387.643-387.671 387.643zM585.143 164.571v109.714c0 4.951-1.808 9.237-5.429 12.857s-7.906 5.429-12.857 5.429h-109.714c-4.953 0-9.239-1.808-12.857-5.429s-5.429-7.906-5.429-12.857v-109.714c0-4.951 1.81-9.237 5.429-12.857s7.904-5.429 12.857-5.429h109.714c4.951 0 9.237 1.808 12.857 5.429s5.429 7.906 5.429 12.857zM521.616 365.714c118.343 0 214.286 88.965 214.286 187.429s-95.943 178.286-214.286 178.286c-173.045 0-208.091-93.714-213.963-171.113 0-7.936 6.729-9.838 15.872-9.838s108.073 0 113.143 0c6.096 0 14.475 0.633 17.554 10.571 0 48.857 135.968 54.953 135.968-7.904 0-31.506-29.717-63.817-68.571-66.194s-82.286-7.607-82.286-54.199c0-13.022 0-25.673 0-44.693 0-19.015 9.717-22.343 27.431-22.343s54.853-0.002 54.853-0.002z\\" transform=\\"rotate(180 0 512) scale(-1 1)\\" />
                                    </svg>
                                  </HelpIcon>
                                </Icon>
                              </div>
                            </GridItem>
                          </div>
                        </Grid>
                        <Text component=\\"h3\\">
                          <h3 data-pf-content={true} className=\\"\\">
                            <Link to=\\"/login\\" style={{...}}>
                              <LinkAnchor style={{...}} href=\\"/login\\" navigate={[Function: navigate]}>
                                <a style={{...}} href=\\"/login\\" onClick={[Function: onClick]}>
                                  <span className=\\"hub-header-login\\">
                                    Login
                                  </span>
                                </a>
                              </LinkAnchor>
                            </Link>
                          </h3>
                        </Text>
                      </div>
                    </PageHeaderTools>
                  </header>
                </PageHeader>
                <Modal variant=\\"small\\" title=\\"Search tips:\\" isOpen={false} onClose={[Function: onClose]} className=\\"\\" aria-label=\\"\\" showClose={true} aria-describedby=\\"\\" aria-labelledby=\\"\\" id={[undefined]} actions={{...}} hasNoBodyWrapper={false} appendTo={[Function: appendTo]} ouiaSafe={true}>
                  <Portal containerInfo={{...}}>
                    <ModalContent variant=\\"small\\" isOpen={false} onClose={[Function: onClose]} className=\\"\\" showClose={true} id={[undefined]} actions={{...}} hasNoBodyWrapper={false} boxId=\\"pf-modal-part-0\\" labelId=\\"pf-modal-part-1\\" descriptorId=\\"pf-modal-part-2\\" title=\\"Search tips:\\" aria-label=\\"\\" aria-describedby=\\"\\" aria-labelledby=\\"\\" ouiaId={[undefined]} ouiaSafe={true} />
                  </Portal>
                </Modal>
              </Memo(wrappedComponent)>
              <main role={[undefined]} id={[undefined]} className=\\"pf-c-page__main\\" tabIndex={-1} aria-label={[undefined]}>
                <Route exact={true} path=\\"/\\" component={[Function: Background]}>
                  <Background history={{...}} location={{...}} match={{...}} staticContext={[undefined]}>
                    <div className=\\"hub-background-image\\" style={{...}}>
                      <TextContent>
                        <div className=\\"pf-c-content\\">
                          <Text component=\\"h1\\" className=\\"hub-background-heading\\">
                            <h1 data-pf-content={true} className=\\"hub-background-heading\\">
                              Welcome to Tekton Hub
                            </h1>
                          </Text>
                          <Text component=\\"h3\\" className=\\"hub-tag-line\\">
                            <h3 data-pf-content={true} className=\\"hub-tag-line\\">
                              Discover, search and share reusable Tasks and Pipelines
                            </h3>
                          </Text>
                        </div>
                      </TextContent>
                    </div>
                  </Background>
                </Route>
                <PageSection>
                  <section className=\\"pf-c-page__main-section\\">
                    <Grid hasGutter={true}>
                      <div className=\\"pf-l-grid pf-m-gutter\\">
                        <GridItem span={12}>
                          <div className=\\"pf-l-grid__item pf-m-12-col\\">
                            <Route exact={true} path=\\"/login\\" component={{...}} />
                            <Route exact={true} path=\\"/:catalog/:kind/:name\\" component={[Function: Details]} />
                          </div>
                        </GridItem>
                        <GridItem span={2}>
                          <div className=\\"pf-l-grid__item pf-m-2-col\\">
                            <Route exact={true} path=\\"/\\" component={[Function: LeftPane]}>
                              <LeftPane history={{...}} location={{...}} match={{...}} staticContext={[undefined]}>
                                <Grid hasGutter={true} className=\\"hub-leftpane\\">
                                  <div className=\\"pf-l-grid pf-m-gutter hub-leftpane\\">
                                    <GridItem span={8}>
                                      <div className=\\"pf-l-grid__item pf-m-8-col\\">
                                        <Sort>
                                          <div className=\\"hub-sort\\">
                                            <Select variant=\\"typeahead\\" typeAheadAriaLabel=\\"Sort By\\" onToggle={[Function: onToggle]} onSelect={[Function: onSelect]} onClear={[Function: clearSelection]} isOpen={false} selections=\\"\\" placeholderText=\\"Sort By\\" className=\\"\\" direction=\\"down\\" toggleId={{...}} isGrouped={false} isPlain={false} isDisabled={false} isCreatable={false} aria-label=\\"\\" aria-labelledby=\\"\\" clearSelectionsAriaLabel=\\"Clear all\\" toggleAriaLabel=\\"Options menu\\" removeSelectionAriaLabel=\\"Remove\\" createText=\\"Create\\" noResultsFoundText=\\"No results found\\" width=\\"\\" onCreateOption={[Function: onCreateOption]} toggleIcon={{...}} onFilter={{...}} customContent={{...}} hasInlineFilter={false} inlineFilterPlaceholderText={{...}} customBadgeText={{...}} inputIdPrefix=\\"\\" menuAppendTo=\\"inline\\" ouiaSafe={true}>
                                              <GenerateId prefix=\\"pf-random-id-\\">
                                                <div className=\\"pf-c-select\\" data-ouia-component-type=\\"PF4/Select\\" data-ouia-safe={true} data-ouia-component-id={0}>
                                                  <SelectToggle id=\\"pf-select-toggle-id-0\\" parentRef={{...}} menuRef={{...}} isOpen={false} isPlain={false} onToggle={[Function: onToggle]} onEnter={[Function (anonymous)]} onClose={[Function (anonymous)]} variant=\\"typeahead\\" aria-labelledby=\\" pf-select-toggle-id-0\\" aria-label=\\"Options menu\\" handleTypeaheadKeys={[Function (anonymous)]} isDisabled={false} hasClearButton={true} className=\\"\\" isActive={false} type=\\"button\\">
                                                    <div className=\\"pf-c-select__toggle pf-m-typeahead\\" onClick={[Function: onClick]} onKeyDown={[Function (anonymous)]}>
                                                      <div className=\\"pf-c-select__toggle-wrapper\\">
                                                        <input className=\\"pf-c-form-control pf-c-select__toggle-typeahead\\" aria-activedescendant={{...}} id=\\"pf-select-toggle-id-0-select-typeahead\\" aria-label=\\"Sort By\\" placeholder=\\"Sort By\\" value=\\"\\" type=\\"text\\" onClick={[Function (anonymous)]} onChange={[Function (anonymous)]} onFocus={[Function (anonymous)]} autoComplete=\\"off\\" disabled={false} />
                                                      </div>
                                                      <button id=\\"pf-select-toggle-id-0\\" aria-labelledby=\\" pf-select-toggle-id-0\\" aria-expanded={false} aria-haspopup=\\"listbox\\" type=\\"button\\" className=\\"pf-c-button pf-c-select__toggle-button pf-m-plain\\" aria-label=\\"Options menu\\" onClick={[Function: onClick]} tabIndex={-1} disabled={false}>
                                                        <CaretDownIcon className=\\"pf-c-select__toggle-arrow\\" color=\\"currentColor\\" size=\\"sm\\" noVerticalAlign={false}>
                                                          <svg style={{...}} fill=\\"currentColor\\" height=\\"1em\\" width=\\"1em\\" viewBox=\\"0 0 320 512\\" aria-labelledby={{...}} aria-hidden={true} role=\\"img\\" className=\\"pf-c-select__toggle-arrow\\">
                                                            <path d=\\"M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z\\" transform=\\"\\" />
                                                          </svg>
                                                        </CaretDownIcon>
                                                      </button>
                                                    </div>
                                                  </SelectToggle>
                                                </div>
                                              </GenerateId>
                                            </Select>
                                          </div>
                                        </Sort>
                                      </div>
                                    </GridItem>
                                    <GridItem>
                                      <div className=\\"pf-l-grid__item\\">
                                        <KindFilter>
                                          <Filter store={{...}} header=\\"Kind\\">
                                            <div>
                                              <Grid>
                                                <div className=\\"pf-l-grid\\">
                                                  <GridItem span={6}>
                                                    <div className=\\"pf-l-grid__item pf-m-6-col\\">
                                                      <Text component=\\"h1\\" className=\\"hub-filter-header\\">
                                                        <h1 data-pf-content={true} className=\\"hub-filter-header\\">
                                                          Kind
                                                        </h1>
                                                      </Text>
                                                    </div>
                                                  </GridItem>
                                                  <GridItem span={3}>
                                                    <div className=\\"pf-l-grid__item pf-m-3-col\\">
                                                      <Button variant=\\"plain\\" aria-label=\\"Clear\\" onClick={[Function: res]}>
                                                        <button onClick={[Function: res]} aria-disabled={false} aria-label=\\"Clear\\" className=\\"pf-c-button pf-m-plain\\" disabled={false} tabIndex={[undefined]} type=\\"button\\" data-ouia-component-type=\\"PF4/Button\\" data-ouia-safe={true} data-ouia-component-id={1}>
                                                          <TimesIcon color=\\"currentColor\\" size=\\"sm\\" noVerticalAlign={false}>
                                                            <svg style={{...}} fill=\\"currentColor\\" height=\\"1em\\" width=\\"1em\\" viewBox=\\"0 0 352 512\\" aria-labelledby={{...}} aria-hidden={true} role=\\"img\\">
                                                              <path d=\\"M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z\\" transform=\\"\\" />
                                                            </svg>
                                                          </TimesIcon>
                                                        </button>
                                                      </Button>
                                                    </div>
                                                  </GridItem>
                                                  <GridItem span={12}>
                                                    <div className=\\"pf-l-grid__item pf-m-12-col\\" />
                                                  </GridItem>
                                                </div>
                                              </Grid>
                                            </div>
                                          </Filter>
                                        </KindFilter>
                                      </div>
                                    </GridItem>
                                    <GridItem>
                                      <div className=\\"pf-l-grid__item\\">
                                        <CatalogFilter>
                                          <Filter store={{...}} header=\\"Catalog\\">
                                            <div>
                                              <Grid>
                                                <div className=\\"pf-l-grid\\">
                                                  <GridItem span={6}>
                                                    <div className=\\"pf-l-grid__item pf-m-6-col\\">
                                                      <Text component=\\"h1\\" className=\\"hub-filter-header\\">
                                                        <h1 data-pf-content={true} className=\\"hub-filter-header\\">
                                                          Catalog
                                                        </h1>
                                                      </Text>
                                                    </div>
                                                  </GridItem>
                                                  <GridItem span={3}>
                                                    <div className=\\"pf-l-grid__item pf-m-3-col\\">
                                                      <Button variant=\\"plain\\" aria-label=\\"Clear\\" onClick={[Function: res]}>
                                                        <button onClick={[Function: res]} aria-disabled={false} aria-label=\\"Clear\\" className=\\"pf-c-button pf-m-plain\\" disabled={false} tabIndex={[undefined]} type=\\"button\\" data-ouia-component-type=\\"PF4/Button\\" data-ouia-safe={true} data-ouia-component-id={2}>
                                                          <TimesIcon color=\\"currentColor\\" size=\\"sm\\" noVerticalAlign={false}>
                                                            <svg style={{...}} fill=\\"currentColor\\" height=\\"1em\\" width=\\"1em\\" viewBox=\\"0 0 352 512\\" aria-labelledby={{...}} aria-hidden={true} role=\\"img\\">
                                                              <path d=\\"M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z\\" transform=\\"\\" />
                                                            </svg>
                                                          </TimesIcon>
                                                        </button>
                                                      </Button>
                                                    </div>
                                                  </GridItem>
                                                  <GridItem span={12}>
                                                    <div className=\\"pf-l-grid__item pf-m-12-col\\" />
                                                  </GridItem>
                                                </div>
                                              </Grid>
                                            </div>
                                          </Filter>
                                        </CatalogFilter>
                                      </div>
                                    </GridItem>
                                    <GridItem>
                                      <div className=\\"pf-l-grid__item\\">
                                        <CategoryFilter>
                                          <Filter store={{...}} header=\\"Category\\">
                                            <div>
                                              <Grid>
                                                <div className=\\"pf-l-grid\\">
                                                  <GridItem span={6}>
                                                    <div className=\\"pf-l-grid__item pf-m-6-col\\">
                                                      <Text component=\\"h1\\" className=\\"hub-filter-header\\">
                                                        <h1 data-pf-content={true} className=\\"hub-filter-header\\">
                                                          Category
                                                        </h1>
                                                      </Text>
                                                    </div>
                                                  </GridItem>
                                                  <GridItem span={3}>
                                                    <div className=\\"pf-l-grid__item pf-m-3-col\\">
                                                      <Button variant=\\"plain\\" aria-label=\\"Clear\\" onClick={[Function: res]}>
                                                        <button onClick={[Function: res]} aria-disabled={false} aria-label=\\"Clear\\" className=\\"pf-c-button pf-m-plain\\" disabled={false} tabIndex={[undefined]} type=\\"button\\" data-ouia-component-type=\\"PF4/Button\\" data-ouia-safe={true} data-ouia-component-id={3}>
                                                          <TimesIcon color=\\"currentColor\\" size=\\"sm\\" noVerticalAlign={false}>
                                                            <svg style={{...}} fill=\\"currentColor\\" height=\\"1em\\" width=\\"1em\\" viewBox=\\"0 0 352 512\\" aria-labelledby={{...}} aria-hidden={true} role=\\"img\\">
                                                              <path d=\\"M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z\\" transform=\\"\\" />
                                                            </svg>
                                                          </TimesIcon>
                                                        </button>
                                                      </Button>
                                                    </div>
                                                  </GridItem>
                                                  <GridItem span={12}>
                                                    <div className=\\"pf-l-grid__item pf-m-12-col\\" />
                                                  </GridItem>
                                                </div>
                                              </Grid>
                                            </div>
                                          </Filter>
                                        </CategoryFilter>
                                      </div>
                                    </GridItem>
                                  </div>
                                </Grid>
                              </LeftPane>
                            </Route>
                          </div>
                        </GridItem>
                        <GridItem span={10} rowSpan={1}>
                          <div className=\\"pf-l-grid__item pf-m-10-col pf-m-1-row\\">
                            <Route exact={true} path=\\"/\\" component={{...}}>
                              <Memo(wrappedComponent) history={{...}} location={{...}} match={{...}} staticContext={[undefined]}>
                                <Spinner className=\\"hub-spinner\\">
                                  <span className=\\"pf-c-spinner pf-m-xl hub-spinner\\" role=\\"progressbar\\" aria-valuetext=\\"Loading...\\">
                                    <span className=\\"pf-c-spinner__clipper\\" />
                                    <span className=\\"pf-c-spinner__lead-ball\\" />
                                    <span className=\\"pf-c-spinner__tail-ball\\" />
                                  </span>
                                </Spinner>
                              </Memo(wrappedComponent)>
                            </Route>
                          </div>
                        </GridItem>
                      </div>
                    </Grid>
                  </section>
                </PageSection>
                <Footer>
                  <Card className=\\"hub-footer-card\\">
                    <article className=\\"pf-c-card hub-footer-card\\" tabIndex={[undefined]} data-ouia-component-type=\\"PF4/Card\\" data-ouia-safe={true} data-ouia-component-id={4}>
                      <Grid>
                        <div className=\\"pf-l-grid\\">
                          <GridItem span={12} className=\\"hub-footer-info\\">
                            <div className=\\"pf-l-grid__item pf-m-12-col hub-footer-info\\">
                              <a href=\\"https://cd.foundation\\" target=\\"_\\">
                                <img src=\\"https://tekton.dev/partner-logos/cdf.png?1609496430000\\" alt=\\"tekton.dev\\" />
                              </a>
                            </div>
                          </GridItem>
                          <GridItem span={12} className=\\"hub-footer-info\\">
                            <div className=\\"pf-l-grid__item pf-m-12-col hub-footer-info\\">
                              <TextContent className=\\"hub-info-color\\">
                                <div className=\\"pf-c-content hub-info-color\\">
                                  <Text component=\\"h3\\">
                                    <h3 data-pf-content={true} className=\\"\\">
                                      Tekton is a
                                       
                                      <Text component=\\"a\\" href=\\"https://cd.foundation\\" target=\\"_\\">
                                        <a href=\\"https://cd.foundation\\" target=\\"_\\" data-pf-content={true} className=\\"\\">
                                          Continuous Delivery Foundation
                                        </a>
                                      </Text>
                                       
                                      project.
                                    </h3>
                                  </Text>
                                </div>
                              </TextContent>
                            </div>
                          </GridItem>
                          <GridItem span={12} className=\\"hub-logo-margin\\">
                            <div className=\\"pf-l-grid__item pf-m-12-col hub-logo-margin\\">
                              <img src=\\"tekton.png?1609496430000\\" alt=\\"Tekton\\" className=\\"hub-logo-size\\" />
                            </div>
                          </GridItem>
                          <GridItem span={12} className=\\"hub-footer-description\\">
                            <div className=\\"pf-l-grid__item pf-m-12-col hub-footer-description\\">
                              <Text>
                                <p data-pf-content={true} className=\\"\\">
                                  © 
                                  2021
                                   The Linux Foundation®. All rights reserved. The Linux Foundation has registered trademarks and uses trademarks. For a list of trademarks of The Linux Foundation, please see our
                                   
                                  <Text component=\\"a\\" href=\\"https://www.linuxfoundation.org/trademark-usage/\\" target=\\"_\\">
                                    <a href=\\"https://www.linuxfoundation.org/trademark-usage/\\" target=\\"_\\" data-pf-content={true} className=\\"\\">
                                      Trademark Usage page
                                    </a>
                                  </Text>
                                  . Linux is a registered trademark of Linus Torvalds.
                                   
                                  <Text component=\\"a\\" href=\\"https://www.linuxfoundation.org/privacy/\\" target=\\"_\\">
                                    <a href=\\"https://www.linuxfoundation.org/privacy/\\" target=\\"_\\" data-pf-content={true} className=\\"\\">
                                      Privacy Policy
                                    </a>
                                  </Text>
                                   
                                  and
                                   
                                  <Text component=\\"a\\" href=\\"https://www.linuxfoundation.org/terms/\\" target=\\"_\\">
                                    <a href=\\"https://www.linuxfoundation.org/terms/\\" target=\\"_\\" data-pf-content={true} className=\\"\\">
                                      Terms of Use
                                    </a>
                                  </Text>
                                   
                                  .
                                </p>
                              </Text>
                            </div>
                          </GridItem>
                        </div>
                      </Grid>
                    </article>
                  </Card>
                </Footer>
              </main>
            </div>
          </Page>
        </Router>
      </BrowserRouter>
       
    </Provider>
  </Memo(wrappedComponent)>
   
</Provider>"
`;
